packages/cdktf-cli/lib/synth.ts

```diff
@@ -51,3 +52,29 @@ export function printAnnotations(stacks: SynthesizedStack[]) {
     );
   }
 }
+
+export function printWarningsForDeprecatedFlags() {
+  const context: Record<string, boolean | undefined> = JSON.parse(
+    process.env[CONTEXT_ENV] || "{}"
+  );
+
+  // We want to print a warning for every deprecated flag that is set to false or not set
+  const deprecatedFlagsInUse = DEPRECATED_FLAGS.filter(
```

Comment:
I'm a bit confused about the naming "in use" and the filtered flags being either not set or set to undefined.

I suspect the underlying problem I'm having in understanding is that there might be two possible situations when a feature flag might get deprecated:
1. The behaviour will be adopted (this is what is currently handled by the code; in this case we want to warn the user that they should set it to `true` to adopt the behaviour, right?)
2. The behaviour will be dropped (this is not yet handled, but in that case we should advise the user to set it to `false` to switch the the behaviour that is going to apply in future versions)

With that in mind, I think we should change the wording in the error message, shouldn't we?

Reply:
Ah, I just read the docs further down. They explain everything 🙈

I'm still a bit confused about how the exact process of "promoting" a feature / feature flag works. It feels that there is a part missing or the wording is just a bit inconsistent for my taste leading to false assumptions (probably around deprecating a feature flag which will turn on the feature by default in the future) but I can't really grasp it.

Reply:
So the process is we introduce something new, on by default if you create a new project. By seeing the flag to false you can get back to the old behavior until we remove the flag. This is helpful to postpone the change needed by a little or to work around bugs on our end